server/apps/users/views.py

```diff
@@ -131,3 +132,100 @@ def get(request):
         serializer = TodoSerializer(todos, many=True)
 
         return Response(serializer.data)
+
+    # Update to'do
+    @staticmethod
+    def put(request, **kwargs):
+        user = request.user
+        todo_id = request.data['todo_id']
+
+        # Get the to'do that user want to update from database and check if to'do in our database has current user as a editor
+        todo = Todo.objects.filter(id=todo_id).first()
+        editors = todo.editors.all()
+
+        # If current user is not an owner or editor for the to'do then don't process the request
+        if user != todo.owner and user not in editors:
+            return Response(OPERATION_NOT_ALLOWED)
+
+        # Set current user as updated by
+        request.data['updated_by'] = user.name
+
+        serializer = TodoSerializer(todo, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return Response(serializer.data)
+
+    # Delete to'do
+    @staticmethod
+    def delete(request, **kwargs):
+        todo_id: int = request.GET['todo-id']
+        user = request.user
+
+        # Get the to'do that user want to update from database and check if to'do in our database has current user as a editor
+        todo = Todo.objects.filter(id=todo_id).first()
+        editors = todo.editors.all()
+
+        # If current user is not an owner or editor for the to'do then don't process the request
+        if user != todo.owner and user not in editors:
+            return Response(OPERATION_NOT_ALLOWED)
+
+        todo.delete()
+
+        return Response(TODO_REMOVED)
+
+
+class TodoUtils:
+    @staticmethod
+    def list_of_todos(request) -> list:
+        todos = TodoView.get(request)
+        todos = json.dumps(todos.data)
+        todos = json.loads(todos)
+        return todos
+
+
+class SearchTodo(generics.RetrieveAPIView, TodoUtils):
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [JSONWebTokenAuthentication]
+
+    @classmethod
```

Comment:
don't use the class method, we need an instance. Override already available methods.

Reply:
Also, @surajpatilvelotio I think the search view itself is incorrect.
We can use drf search filter

Reply:
https://www.django-rest-framework.org/api-guide/filtering/#searchfilter

Reply:
Updated